src/main/java/org/jenkinsci/plugins/matrixauth/integrations/casc/MatrixAuthorizationStrategyConfigurator.java

```diff
@@ -29,47 +34,43 @@
         return AuthorizationStrategy.class;
     }
 
-
     @Override
     @Nonnull
     public Set<Attribute<T, ?>> describe() {
-        return new HashSet<>(Arrays.asList(
-                new MultivaluedAttribute<T, String>("permissions", String.class)
+        return new HashSet<>(Collections.singletonList(
+                new MultivaluedAttribute<T, PermissionEntryForCasc>("permissions", PermissionEntryForCasc.class)
                         .getter(MatrixAuthorizationStrategyConfigurator::getPermissions)
-                        .setter(MatrixAuthorizationStrategyConfigurator::setPermissions),
-
-                // support old style configuration options
-                new MultivaluedAttribute<T, String>("grantedPermissions", String.class)
-                        .getter(unused -> null)
-                        .setter(MatrixAuthorizationStrategyConfigurator::setPermissionsDeprecated)
+                        .setter(MatrixAuthorizationStrategyConfigurator::setPermissions)
         ));
     }
 
     /**
      * Extract container's permissions as a List of "TYPE:PERMISSION:sid"
      */
-    public static Collection<String> getPermissions(AuthorizationContainer container) {
+    public static Collection<PermissionEntryForCasc> getPermissions(AuthorizationContainer container) {
         return container.getGrantedPermissionEntries().entrySet().stream()
                 .flatMap( e -> e.getValue().stream()
-                        .map(v -> v.getType().toPrefix() + e.getKey().group.getId() + "/" + e.getKey().name + ":" + v.getSid()))
+                        .map(v -> {
+                            PermissionEntryForCasc entry = new PermissionEntryForCasc(e.getKey().group.getId() + "/" + e.getKey().name);
+                            if (v.getType().equals(AuthorizationType.USER)) {
+                                entry.setUser(v.getSid());
+                                return entry;
+                            } else if (v.getType().equals(AuthorizationType.GROUP)) {
+                                entry.setGroup(v.getSid());
+                                return entry;
+                            }
+                            throw new IllegalStateException("Ambiguous users are not supported: " + v.getSid());
```

Comment:
Causes a failure to export if there's any `either` types
![image](https://user-images.githubusercontent.com/21194782/147081598-0b9a3cfb-0bee-4c3d-9273-b61cc9d99779.png)

Could:
- Add support for either with a log message maybe?
- Filter either out
- or just have it error like this? not very nice UX though, thoughts?